internal/pkg/template/api/extension/token/token.go

```diff
@@ -0,0 +1,268 @@
+				for _, param := range path.Get.Parameters {
+					filtered := make([]*openapiv2.Parameter, 0)
+					if param.Name != headerName {
+						filtered = append(filtered, param)
+					}
+					path.Get.Parameters = filtered
+				}
+
+				// Normalize security name in reference
+				for _, item := range path.Get.Security {
+					for key, value := range item {
+						if key == originalSecurityName {
+							delete(item, key)
+							item[modifiedSecurityName] = value
+						}
+					}
+				}
+
+				// Rename the name in the security definition, see below
+				modifiedSecurities[originalSecurityName] = modifiedSecurityName
+			}
+		}
+
+		// Iterate all securities
+		securities := data.SecurityDefinitions
+		for key, value := range securities {
+			// Normalize security name in definition
+			if modifiedKey, ok := modifiedSecurities[key]; ok {
+				delete(securities, key)
+				securities[modifiedKey] = value
+			}
+		}
+	}
+}
+
+func modifyOpenApiV3(roots []eval.Root, data *openapiv3.OpenAPI) {
+	modifiedSecurities := make(map[string]string)
+
+	for _, path := range data.Paths {
+		// Iterate all endpoints
+		if path.Get != nil {
+			headerName, originalSecurityName, modifiedSecurityName, found := resolveNaming(roots, path.Get.Tags)
+			if !found {
+				continue
+			}
+
+			// Skip token header definition that is unnecessary
+			for _, param := range path.Get.Parameters {
+				filtered := make([]*openapiv3.ParameterRef, 0)
+				if param.Value.Name != headerName {
+					filtered = append(filtered, param)
+				}
+				path.Get.Parameters = filtered
+			}
+
+			// Normalize security name in reference
+			for _, item := range path.Get.Security {
+				for key, value := range item {
+					if key == originalSecurityName {
+						delete(item, key)
+						item[modifiedSecurityName] = value
+					}
+				}
+			}
+
+			// Rename the name in the security definition, see below
+			modifiedSecurities[originalSecurityName] = modifiedSecurityName
+		}
+
+		// Iterate all securities
+		securities := data.Components.SecuritySchemes
+		for key, value := range securities {
+			// Normalize security name in definition
+			if modifiedKey, ok := modifiedSecurities[key]; ok {
+				delete(securities, key)
+				securities[modifiedKey] = value
+			}
+		}
+	}
+}
+
+func resolveNaming(roots []eval.Root, tags []string) (string, string, string, bool) {
+	var schemaName, headerName, originalSecurityName, modifiedSecurityName string
+
+	// Get naming from the service, by OpenApi tags (tag = service name)
+	for _, root := range roots {
+		root.WalkSets(func(set eval.ExpressionSet) error {
+			for _, item := range set {
+				if api, ok := item.(*expr.APIExpr); ok {
+					for _, tag := range tags {
+						if httpService := api.HTTP.Service(tag); httpService != nil {
+							service := httpService.ServiceExpr
+							v1, f1 := service.Meta.Last(MetaKeySchemeName)
+							v2, f2 := service.Meta.Last(MetaKeyTokenHeader)
+							if f1 && f2 {
```

Comment:
Ziskame potrebne informacie z metadata `service`. Vid riadok `63` tento subor, spravil som to trocha viac univerzalne ako bolo nutne, ... kedze ma zaujimalo ako sa to da v Goa spravit :) Namiesto metadat to mohli byt konstanty, kedze mame iba jednu service.